sympy/physics/control/lti.py

```diff
@@ -744,24 +783,32 @@ class Series(Basic):
     See Also
     ========
 
-    Parallel, TransferFunction, Feedback
+    MIMOSeries, Parallel, TransferFunction, Feedback
 
     """
     def __new__(cls, *args, evaluate=False):
-        if not all(isinstance(arg, (TransferFunction, Parallel, Series)) for arg in args):
-            raise TypeError("Unsupported type of argument(s) for Series.")
-
-        obj = super().__new__(cls, *args)
-        obj._var = None
-        for arg in args:
-            if obj._var is None:
-                obj._var = arg.var
-            elif obj._var != arg.var:
-                raise ValueError("All transfer functions should use the same complex"
-                    " variable of the Laplace transform.")
-        if evaluate:
-            return obj.doit()
-        return obj
+        if len(args) == 0:
+            raise ValueError("Atleast 1 argument must be passed.")
+
+        if not all(isinstance(arg, LinearTimeInvariant) for arg in args):
+            raise TypeError("All arguments must be of type LinearTimeInvariant.")
+
+        type_set = {arg.is_SISO for arg in args}
+        if len(type_set) != 1:
+            raise TypeError("Invalid system. `args` passed contain SISO as well "
+                "as MIMO elements.")
+        var_set = {arg.var for arg in args}
+        if len(var_set) != 1:
+            raise ValueError("All transfer functions should use the same complex variable"
+                f" of the Laplace transform. {len(var_set)} different values found.")
+
+        if list(type_set)[0]:
+            obj = super().__new__(cls, *args)
+            obj._is_SISO = True
+        else:
+            raise TypeError("MIMO systems are not allowed in `Series`. Use `MIMOSeries` instead.")
```

Comment:
I don't really see the point in checking for a mix of MIMO and SISO, then checking for any MIMO. Why not just check SISO from the beginning?
```suggestion
        if not all(arg.is_SISO for arg in args):
            raise TypeError("MIMO systems are not allowed in `Series`. Use `MIMOSeries` instead.")

        var_set = {arg.var for arg in args}
        if len(var_set) != 1:
            raise ValueError("All transfer functions should use the same complex variable"
                f" of the Laplace transform. {len(var_set)} different values found.")

        obj = super().__new__(cls, *args)
        obj._is_SISO = True
```

Reply:
There are lots of places that do this:
```python
if not all(isinstance(arg, LinearTimeInvariant) for arg in args):
    raise ...

if not all(arg.is_SISO for arg in args):
    raise ...

var_set = {arg.var for arg in args}
if len(var_set) != 1:
    raise ...
```
Now that SISO and MIMO are always distinct classes it would be possible to make a common superclass for SISO like `SISOLinearTimeInvariant`. Then the first two tests could be combined like:
```python
if not all(isinstance(arg, SISOLinearTimeInvariant) for arg in args):
    raise...
```
Also since this checking is quite common it should be factored out into a helper in the super class so it looks like this:
```python
class LinearTimeInvariant:

    @classmethod
    def _check_args(cls, args):
        if not all(isinstance(arg, cls._clstype) for arg in args):
            raise ...        
        var_set = {arg.var for arg in args}
        if len(var_set) != 1:
            raise ...
        (var,) = var_set
        return var  # does the caller need this?

    @property
    def is_SISO(self):
        return self._is_SISO


class SISOLinearTimeInvariant(LinearTimeInvariant):
    _is_SISO = True


class MIMOLinearTimeInvariant(LinearTimeInvariant
    _is_SISO = False


SISOLinearTimeInvariant._clstype = SISOLinearTimeInvariant
MIMOLinearTimeInvariant._clstype = MIMOLinearTimeInvariant


class Series(SISOLinearTimeInvariant):

    def __new__(cls, *args):
        if not args:
            raise ... # maybe this could be handled in _check_args as well?
        var = cls._check_args(args)
        return super().__new__(cls, *args)
```